src/tree/mod.rs

```diff
@@ -129,11 +111,32 @@ where
         &self,
         idx: u64,
     ) -> io::Result<Option<PoseidonBranch>> {
-        Ok(self.inner.get(idx)?.map(|ref branch| {
-            let mut pbranch: PoseidonBranch = branch.into();
-            pbranch.extend(self.branch_depth as usize);
-            pbranch
-        }))
+        // Try to get the PoseidonBranch from the tree.
+        let mut pbranch: PoseidonBranch =
+            self.inner.get(idx)?.map(|ref branch| branch.into()).ok_or(
```

Comment:
Since you're calling a function, use lazy evaluation (`ok_or_else`).

Reply:
Totally correct!